tests/test_S_OPT.C

```diff
@@ -0,0 +1,375 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: This source file is a test runner that uses the Google Test
+// Framework to run unit tests on the CAROM::S_OPT class.
+
+#include <iostream>
+
+#ifdef CAROM_HAS_GTEST
+#include<gtest/gtest.h>
+#include <mpi.h>
+#include "hyperreduction/S_OPT.h"
+#include "linalg/Matrix.h"
+#include "utils/CSVDatabase.h"
+#define _USE_MATH_DEFINES
+#include <cmath>
+
+/**
+ * Simple smoke test to make sure Google Test is properly linked
+ */
+TEST(GoogleTestFramework, GoogleTestFrameworkFound) {
+    SUCCEED();
+}
+
+TEST(S_OPTSerialTest, Test_S_OPT)
+{
+    // Get the rank of this process, and the number of processors.
+    int mpi_init, d_rank, d_num_procs;
+    MPI_Initialized(&mpi_init);
+    if (mpi_init == 0) {
+        MPI_Init(nullptr, nullptr);
+    }
+
+    MPI_Comm_rank(MPI_COMM_WORLD, &d_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &d_num_procs);
+
+    int num_total_rows = 1000;
```

Comment:
There are several data files for this test, which appear to come from a matlab example. Having 1000 entries in 5 files is not too bad, but it would be ideal to avoid adding data files like this to the repo. Can similar data files be generated by the test? Or can they be smaller (e.g. 100 entries)?